Zeeguu Reader/Pods/Zeeguu-API-iOS/ZeeguuAPI/ZeeguuAPI/Article.swift

```diff
@@ -122,13 +126,79 @@ public class Article: CustomStringConvertible, Equatable {
 		self.feed = feed
 		self.title = title;
 		self.url = url;
-		self.date = date;
+		
+		let formatter = NSDateFormatter()
+		formatter.locale = NSLocale(localeIdentifier: "EN-US")
+		formatter.dateFormat = "EEE, dd MMMM y HH:mm:ss Z"
+		
+		let date = formatter.dateFromString(date)
+		self.date = date!
 		self.summary = summary
+		self.isRead = false
+		self.isStarred = false
+	}
+	
+	/**
+	Construct a new `Article` object from the data in the dictionary.
+	
+	- parameter dictionary: The dictionary that contains the data from which to construct an `Article` object.
+	*/
+	@objc public required init?(dictionary dict: [String: AnyObject]) {
+		var savedDate = dict["date"] as? NSDate
+		if let date = dict["date"] as? String {
+			let formatter = NSDateFormatter()
+			formatter.locale = NSLocale(localeIdentifier: "EN-US")
+			formatter.dateFormat = "EEE, dd MMMM y HH:mm:ss Z"
+			
+			let date = formatter.dateFromString(date)
+			savedDate = date
+		}
+		guard let feed = dict["feed"] as? Feed,
+			title = dict["title"] as? String,
+			url = dict["url"] as? String,
+			date = savedDate,
+			summary = dict["summary"] as? String else {
+				return nil
+		}
+		self.feed = feed
+		self.title = title
+		self.url = url
+		self.date = date
+		self.summary = summary
+		self.isRead = dict["isRead"] as? Bool == true
+		self.isStarred = dict["isStarred"] as? Bool == true
+		self.imageURL = dict["imageURL"] as? String
+		self.image = dict["image"] as? UIImage
+		self.contents = dict["contents"] as? String
+		if let difficulty = dict["difficulty"] as? String {
+			self.difficulty = ArticleDifficulty(rawValue: difficulty)
+		}
 	}
 	
 	// MARK: Methods -
 	
 	/**
+	The dictionary representation of this `Article` object.
+	
+	- returns: A dictionary that contains all data of this `Article` object.
+	*/
+	@objc public func dictionaryRepresentation() -> [String: AnyObject] {
+		var dict = [String: AnyObject]()
+		dict["feed"] = self.feed
+		dict["title"] = self.title
+		dict["url"] = self.url
+		dict["date"] = self.date
+		dict["summary"] = self.summary
+		dict["isRead"] = self.isRead
+		dict["isStarred"] = self.isStarred
```

Comment:
No, I just added it, so I don't need to edit it again, but I don't think it is good to do now, as it requires more than just a star button :)